packages/react/src/account/useSetAccountMetadata.ts

```diff
@@ -0,0 +1,40 @@
+import type {
+  OperationHandler,
+  SetAccountMetadataRequest,
+  SigningError,
+  TransactionIndexingError,
+  UnauthenticatedError,
+  UnexpectedError,
+  ValidationError,
+} from '@lens-protocol/client';
+import { setAccountMetadata } from '@lens-protocol/client/actions';
+
+import type { TxHash } from '@lens-protocol/types';
+import { type UseAsyncTask, useAuthenticatedAsyncTask } from '../helpers';
+
+export type UseSetAccountMetadataArgs = {
+  handler: OperationHandler;
+};
+
+/**
+ * Set account metadata.
+ *
+ * @alpha This is an alpha API and may be subject to breaking changes.
+ */
+export function useSetAccountMetadata(
+  args: UseSetAccountMetadataArgs,
+): UseAsyncTask<
+  SetAccountMetadataRequest,
+  TxHash,
+  | SigningError
+  | ValidationError
+  | TransactionIndexingError
+  | UnauthenticatedError
+  | UnexpectedError
+> {
+  return useAuthenticatedAsyncTask((sessionClient, request) =>
+    setAccountMetadata(sessionClient, request)
+      .andThen(args.handler)
+      .andThen(sessionClient.waitForTransaction),
```

Comment:
It's pretty difficult to do any sort of optimistic updates since we just receive the metadata URI from the caller. But we could do an effort for updating the cache at the end. We can re-fetch the account with `fetchAccount`, in doing so the cache will update and any rendered hook (e.g. `useAccount`) returning the same Account should re-render.

Also, unless we have a use case, let's not return what we have (e.g. `TxHash`). Sometimes is better to return Void and potentially change it later, than return something we don't know if useful or not and then be stuck in not being able to change that something. Changing Void into something is not a breaking change, but changing `TxHash` into something else, is a breaking change.